### PR TITLE
Update rook version to v1.12.2

### PR DESCRIPTION
**Pull request by malayparida2000**

### CODE REVIEW

File: controllers/storagecluster/cephcluster.go [hunk 1]
```diff
@@ -1026,13 +1026,13 @@ func generateStretchClusterSpec(sc *ocsv1.StorageCluster) *rookCephv1.StretchClu
 		if zone == sc.Spec.NodeTopologies.ArbiterLocation {
 			continue
 		}
-		stretchClusterSpec.Zones = append(stretchClusterSpec.Zones, rookCephv1.StretchClusterZoneSpec{
+		stretchClusterSpec.Zones = append(stretchClusterSpec.Zones, rookCephv1.MonZoneSpec{
```
Comment (umangachapagain) [inline controllers/storagecluster/cephcluster.go:1029]: @travisn is this change ok?

Comment (malayparida2000) [reply controllers/storagecluster/cephcluster.go:1029]: This was necessary due to this https://github.com/rook/rook/pull/12384/commits/2231d73259707e082c24419bf68e4382444e3c6f#diff-46a2e53ea9b825b7c3b65113b63a133da71eb0cd2a2c0e33012c224d02a8eaa1

**Review by umangachapagain — CHANGES_REQUESTED** [review]
Squash the first 2 commits for go deps. It's the same thing so better to keep it together.

File: controllers/storagecluster/cephcluster.go (hunk 1 above)
Comment (malayparida2000) [reply controllers/storagecluster/cephcluster.go:1029]: Seems like a rename only, @travisn can confirm

**Comment by malayparida2000** [comment]
> Squash the first 2 commits for go deps. It's the same thing so better to keep it together.

done

File: controllers/storagecluster/cephcluster.go (hunk 1 above)
Comment (travisn) [reply controllers/storagecluster/cephcluster.go:1029]: Correct, only the type name changed in a refactoring, but the json representation is the same and is not expected to affect upgrades

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**